test/framework/src/test/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertionsTests.java

```diff
@@ -52,4 +59,172 @@ public void writeTo(StreamOutput out) throws IOException {
             throw new IllegalArgumentException("Not supported.");
         }
     }
+
+    public void testAssertXContentEquivalent() throws IOException {
```

Comment:
similar comment to one that I previously left. with this randomized json, is it necessary to use assertToXContentEquivalent to do the comparison? I think it is because we shuffle now, but maybe we can also make this test leverage the the behaviour of assertObjectEquals with byte[] ?